static/app/components/events/interfaces/spans/spanTreeModel.tsx

```diff
@@ -683,29 +685,44 @@ class SpanTreeModel {
     addTraceBounds: (bounds: TraceBound) => void;
     removeTraceBounds: (eventSlug: string) => void;
   }) =>
-    action('toggleEmbeddedChildren', (props: {eventSlug: string; orgSlug: string}) => {
-      this.showEmbeddedChildren = !this.showEmbeddedChildren;
-      this.fetchEmbeddedChildrenState = 'idle';
+    action(
+      'toggleEmbeddedChildren',
+      (orgSlug: string, transactions: QuickTraceEvent[]) => {
+        this.showEmbeddedChildren = !this.showEmbeddedChildren;
+        this.fetchEmbeddedChildrenState = 'idle';
+
+        if (!this.showEmbeddedChildren) {
+          if (this.embeddedChildren.length > 0) {
+            this.embeddedChildren.forEach(child => {
+              removeTraceBounds(child.generateTraceBounds().spanId);
+            });
+          }
+        }
 
-      if (!this.showEmbeddedChildren) {
-        if (this.embeddedChildren.length > 0) {
+        if (this.showEmbeddedChildren) {
+          if (this.embeddedChildren.length === 0) {
+            Promise.all(
```

Comment:
I'm not familiar with Mobx, but does this need a `return`? There used to be a `return` here!

Reply:
Good catch, I meant to return `Promise.all` here, although surprisingly it still works 🤔